woocommerce/woocommerce-ios
# Lightweight Storefront: Update theme setting screen to display theme list

Closes: #11337 


## Description

This PR integrates the API to fetch suggested themes and adds the themes carousel to the theme setting screens. Changes include:
- `ThemeSettingView`: Added theme carousel view in place.
- `ThemeSettingViewModel`: added a reference to the view model of the carousel view to load themes after getting the current theme.
- Added a view model for `ThemesCarouselView` and integrated the themes API for loading real data.
- Updated the `ThemesCarouselView` with different states based on the load request.
- Updated the Settings screen to display items in the correct order following the design.

## Testing instructions

- Log in to a WPCom store and navigate to the Menu tab.
- Select Settings. If the store hasn't completed onboarding tasks, notice that the Themes item is below the option to hide the onboarding task list.
- Select Themes. Confirm that the screen displays the suggested themes as filtered in `WordPressThemeRemote`. If your store has one of the suggested theme, the theme should not be displayed on the list.

## Screenshots




https://github.com/woocommerce/woocommerce-ios/assets/5533851/a784460a-b706-4934-a368-c9262e3f0099




---
